src/sentry.js

```diff
@@ -1,11 +1,55 @@
 /* @flow strict-local */
-import { Sentry } from 'react-native-sentry';
-
+import * as Sentry from '@sentry/react-native';
 import config from './config';
 
-if (config.enableSentry) {
-  Sentry.config(config.sentryKey, {
-    deactivateStacktraceMerging: true,
-    ignoreErrors: ['Network request failed'],
-  }).install();
-}
+const isSentryUnsafe = (): string | null => {
+  // Under no circumstances should Jest ever try to initialize Sentry.
+  //   [1] https://jestjs.io/docs/en/24.0/getting-started.html#using-babel
+  //   [2] https://stackoverflow.com/a/52231746
+  if (process.env.NODE_ENV === 'test' || process.env.JEST_WORKER_ID !== undefined) {
+    return 'Sentry should not be initialized while testing!';
+  }
+
+  // Running Sentry in debug mode is technically *safe*, although it's probably
+  // not *useful*. (The build process doesn't ordinarily create bundles or .map
+  // files in debug mode, so you'll probably get nonsensical stack traces.)
+  /* if (process.env.NODE_ENV === 'development') {
```

Comment:
I'm a bit confused by this pair of comments: why is it unsafe in tests, while merely not useful in debug mode?

What happens if it runs in tests? I'm imagining that it'd put some noise data in the Sentry console. But that's the same thing that would happen in a dev build, right?

Reply:
That's one scenario. Another, arguably slightly better scenario is that the calls raise (mildly cryptic) exceptions because Sentry isn't properly mocked.

Jest has no good reason to even *try* to initialize Sentry, no matter what the value of `sentryKey` is. (And because `initSentry` is invoked as a side effect of importing this file, it's possible that we might unknowingly do so when testing something else.) There is some support from Sentry for Jest-mocking... but until and unless we actually have a specific reason to jump through those hoops, I'd rather just yell.

That said, the other side was still wrong. As the comment below notes, it's acceptable and even normal to call `initSentry` while in debug mode; it should always have a null `sentryKey` in this case (unless, perhaps, *&lt;more complicated condition&gt;*), but the attempt will be made, just as is done in release.

And of course, "safe" was never the right word here. I've renamed it to "preventNoise", and simplified the call site.

Reply:
Addendum: if there are still issues with this commit, I'd rather just drop it than hold up the remainder of this commit. Additional checks are generally nice, but we've done fine without these so far.

Reply:
Very helpful, thanks!